ophyd/status.py

```diff
@@ -127,25 +184,118 @@ def _settled(self):
         """Hook for when status has completed and settled"""
         pass
 
-    def _settle_then_run_callbacks(self, success=True):
-        # wait until the settling time is done to mark completion
-        if self.settle_time > 0.0:
-            time.sleep(self.settle_time)
-
+    def _run_callbacks(self):
+        """
+        Set the Event and run the callbacks.
+        """
+        if self.timeout is None:
+            timeout = None
+        else:
+            timeout = self.timeout + self.settle_time
+        if not self._settled_event.wait(timeout):
```

Comment:
I worry a bit that improperly used status objects will leak a thread forever (assuming `timeout=None`), but I'm not sure what could/should be done about that.

This is ostensibly a small issue with no known offenders. That said, bluesky/ophyd processes tend to run for weeks on end for scanning beamlines, where leaking a thread an hour (for example) could add up quickly.